hbase-protocol-shaded/src/main/protobuf/Admin.proto

```diff
@@ -287,6 +287,8 @@ message SlowLogResponseRequest {
   optional string client_address = 3;
   optional string user_name = 4;
   optional uint32 limit = 5 [default = 10];
+  optional bool is_slow_log = 6 [default = false];
```

Comment:
use an enum? OnlineLogType for ex.. Also, you don't need to have two separate RPCs if you pass this "type" as a param to the RPC call (we can potentially have other "types" in the future in which case we don't want to add a new set of RPCs).

Reply:
Another thought..why not implement this type as a LogQueryFilter? that makes it much simpler right?

Reply:
IMHO having separate shell command and Admin API should be good and clean for operators rather than having "type" as a filter. I agree on implementing enum for "type" on server side including this proto file. We can have filter logic at server side but having separate API for client would be better because all filters that we want to provide to client should be related to actual data filters - client, user, region, table etc rather than type.

Most of the time, user would not use filter so we don't want them to use filter just for log type.
`get_slowlog_responses '*'` and `get_largelog_responses '*'` might be mostly used (without filters).

Reply:
> IMHO having separate shell command and Admin API should be good and clean for operators rather than having "type" as a filter. 

Umm.. shell command is fine, probably makes it easier for operators...whats the use of having a separate Admin API? All of these requests fall under the same "get_log" category, so probably good to logically group them into same API (which takes Type parameter) and avoid code bloat?

Reply:
I believe we follow 1:1 shell command to Admin API right? This is to ensure operator using shell commands and Java/Thrift clients using Admin API have similar experience. No?

Reply:
They still have the same experience, its just what each client does with the API. For example, there are 10 (or more) ways of creating a table from shell but only 3 corresponding create admin APIs. All these 10 map to one of these 3 APIs. Java/Thrift clients for this API build the TableDescriptor and use one of these 3 API calls. It is somewhat a similar case here. At least this is my understanding.

Reply:
Hmm yeah makes sense. I am not too strong of an opinion to keep 1:1 for slow and large logs. Just thought of keeping them same but yes having single API with LogQueryFilter to differentiate between slow and large logs seems better approach.